tensorflow/lite/delegates/nnapi/java/src/main/native/nnapi_delegate_jni.cc

```diff
@@ -21,10 +21,28 @@ limitations under the License.
 extern "C" {
 #endif  // __cplusplus
 
+using namespace tflite;
+
 JNIEXPORT jlong JNICALL
-Java_org_tensorflow_lite_nnapi_NnApiDelegate_createDelegate(JNIEnv* env,
-                                                            jclass clazz) {
-  return reinterpret_cast<jlong>(tflite::NnApiDelegate());
+Java_org_tensorflow_lite_nnapi_NnApiDelegate_createDelegate(
+    JNIEnv* env, jclass clazz, jint preference, jstring accelerator_name,
+    jstring cache_dir, jstring model_token) {
+
+  StatefulNnApiDelegate::Options options = StatefulNnApiDelegate::Options();
+  options.execution_preference =
+      (StatefulNnApiDelegate::Options::ExecutionPreference)preference;
+  options.accelerator_name = env->GetStringUTFChars(accelerator_name, NULL);
+  options.cache_dir = env->GetStringUTFChars(cache_dir, NULL);
+  options.model_token = env->GetStringUTFChars(model_token, NULL);
+
+  return reinterpret_cast<jlong>(new StatefulNnApiDelegate(options));
```

Comment:
You also need to free the UTFChars if they were created, e.g.,

std::unique_ptr<StatefulNnApiDelegate> delegate(new StatefulNnApiDelegate(options));
if (options.accelerator_name) {
  env->ReleaseStringUTFChars(accelerator_name, options.accelerator_name);
}